### DISCUSSION TITLE
Yash development status and roadmap

### DISCUSSION THREAD

**Discussion by magicant** [magicant/yash#92] [General]
As of December 2024, I'm working on yash to implement the requirements of the POSIX standard, which was updated this year as POSIX.1-2024. It will take me months to finish, considering the amount of work required as listed in #84.

POSIX.1-2024 revised the semantics of command execution, requiring the shell to handle arbitrary byte sequences that may not represent valid characters during command line expansion. Since most of the yash codebase is written in terms of valid wide character strings, a massive rewrite would be required to implement the new behavior. It is likely that the new behavior will only be partially supported, or possibly not implemented at all.

As I've been spending more and more time on [yash-rs](https://github.com/magicant/yash-rs), yash is unlikely to get new features or other updates except for serious bug fixes (if any) and updates for POSIX support as mentioned above. However, the yash repository will not be archived until yash-rs develops mature enough as an interactive shell for daily use, which will be years from now.

**Comment by unrealapex** [comment]
I find Yash to be very useful to me and I would like to help sustain its development in any way I can. If you have any low risk tasks in #84 you think I could work on, please let me know. Writing patches for Yash has been a fun way to get out of my comfort zone and learn C and more about POSIX shell.

**Reply by magicant** [reply]
@unrealapex Great to hear that! I added some good first issues as subtasks of #84 that you might want to check out.
